README.md

```diff
@@ -46,52 +46,92 @@ For each module, the registry contains a JSON file with the following structure:
-
-- The "name" property represents the name of the module.
-- The "details" property is structured as an array of strings. Each array entry represents a paragraph (which will be rendered in a UI).
-- The "links" property should have at least one link to the module's source code.
-- The "parameters" property is used in order to dynamically fetch and display data from module instances.
-  - The "label" property is used to display the name/description of each parameter.
-  - The "functionName" property represents the name of the function from which the parameter should be retrieved. The function should be a view/pure function with no input and one output (not a tuple, but can be an array).
-  - The "displayType" property is used in order to display a proper UI component for the parameter. For example, displaying a date for a parameter representing a timestamp. The supported types are currently:
-    - "default" - Infers automatically the UI component for the value
-    - "timestamp" - Unix timestamp, will be presented as a Date component
-    - "hat" - hat component
-    - "token" - Token component
-    - "seconds" - Amount of time denominated in seconds
-    - "amountWithDecimals" - For token amounts, takes into account the token's decimals
-- There should be at least one field on the "type" property which is set to "true". A module might serve as more than one type, in which case there will be more than one field set to "true".
-- For each chain provided in "deployments", there should be a deployed implementation contract with an address matching the provided "implementationAddress" property and with an ABI
-  matching the "abi" property.
-- For both the immutable and mutable arguments, their order should match the order of the arguments as provided to the module.
-- The "example" fields will be used to automatically test the module's deployment.
+_Note that arrays in the object above contain one example entry._
+
+**Here are some useful notes on its expected structure:**
+
+- `details` - Structured as an array of strings, each array entry represents a paragraph. The property's purpose is to conatain the module's description.
+- `links` - Property's purpose is to include any relevant links about this module and should have at least one link to the module's source code.
+- `parameters` Property is used in order to dynamically fetch and display data from module instances. Using this property, the module creator can choose which module fields are relevant for display.
+  - `label` - Used to display the name/description of each parameter.
+  - `functionName` - Name of the function from which the parameter should be retrieved. The function should be a view/pure function with no inputs and only one output (an array is also considered as one output).
+  - `displayType` - Used in order to display a proper UI component for the parameter. For example, displaying a date for a parameter representing a timestamp. The known supported types are currently:
```

Comment:
The concept of displayTypes needs to be documented more fully. For example, its not clear here that front ends are the determinants of which displayTypes are supported.